File: cems-SPA/src/app/dashboard/csharp/models/DotnetHttpContext.model.ts
import { DotnetRequestPropertiesModel } from './DotnetRequestProperties.model';
import { DotnetConnectionInfoModel } from './DotnetConnectionInfo.model';

export class DotnetHttpContextModel {

    public request: DotnetRequestPropertiesModel = null;
    public connection: DotnetConnectionInfoModel = null;
    public user: string = null;

    constructor(dotnetHttpContextModelObj) {
        this.request = new DotnetRequestPropertiesModel(dotnetHttpContextModelObj.request);
        this.connection = new DotnetConnectionInfoModel(dotnetHttpContextModelObj.connection);
        this.user = dotnetHttpContextModelObj.user;
    }

}

File: cems.API/Models/dotnet/DotnetHttpContextModel.cs
namespace cems.API.Features.LogEndpoint.dotnet.Models
{
    public class DotnetHttpContextModel 
    {
        public DotnetRequestPropertiesModel Request { get; set; }
        public DotnetConnectionInfoModel Connection { get; set; }
        public string User { get; set; }

        public DotnetHttpContextModel()
        {
            Request = new DotnetRequestPropertiesModel();
            Connection = new DotnetConnectionInfoModel();
        }
    }
}